lib/internal/fs/promises.js

```diff
@@ -868,6 +868,14 @@ async function mkdir(path, options) {
 
 async function readdirRecursive(originalPath, options) {
   const result = [];
+  const skipSymlinkDir = options.skipSymlinkDir || false;
+
+  if (skipSymlinkDir) {
+    const stats = await lstat(originalPath);
```

Comment:
I'm no CPP expert, but benchmark-wise, wouldn't this call make us get the file twice? Is there a CPP way to achieve this same result?

> Disclaimer: I'm not a CPP expert (or even a novice)

Reply:
I haven't find an effective way to acquire `stat` in cpp way directly to avoid this dilemma, it would be great if anyone could help, and this will only affect efficiency when using `skipSymlinkDir` to skip symbolic link dirs.